src/pkg/clients/Service_discovery.go

```diff
@@ -38,30 +40,44 @@ func NewServiceDiscovery(host, user, password string) (ServiceDiscovery, error)
 
 	// Create a Consul agent client
 	agent := client.Agent()
-	service := serviceDiscovery{agent: agent}
+	service := serviceDiscovery{agent: agent, transformersAddressesList: map[string][]string{}}
+	if err := service.getTransformationServices(); err != nil {
+		return nil, err
+	}
 
-	return service, nil
+	return &service, nil
 }
 
 // Get all available instances of transformation services
-func (s serviceDiscovery) GetTransformationServices() ([]*TransformerInfos, error) {
+func (s *serviceDiscovery) getTransformationServices() error {
 	services, err := s.agent.ServicesWithFilter("transformer in Service")
 	if err != nil {
-		return nil, err
+		return err
+	}
+	transformerServices := s.parseTransformerInfos(services)
+	s.transformersAddressesList = map[string][]string{}
+	for _, service := range transformerServices {
+		name := strings.Split(service.Name, "-")[0]
+		address := service.Address + ":" + service.Port
+		s.transformersAddressesList[name] = append(s.transformersAddressesList[name], address)
```

Comment:
Since we use map[string][]string, i'm not sure we need anymore the TransformerInfos structure.
We could fill the map directly into parseTransformerInfos